frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -311,3 +315,20 @@ export const pipelineRunDuration = (run: PipelineRun): string => {
     : new Date().getTime() - start;
   return formatDuration(duration);
 };
+
+export const updateServiceAccount = (secret, serviceAccount): Promise<K8sResourceKind> => {
+  serviceAccount.secrets = [...serviceAccount.secrets, ...[secret]];
```

Comment:
Couple things are wrong here... 

You're doing an necessary wrap:
```suggestion
  serviceAccount.secrets = [...serviceAccount.secrets, secret];
```

And you're mutating a provided variable. Recommend you don't ever do that just on principle. It can have unexpected side-effects higher up... obv not here, because it's a result of a get, but still good practice not to do it. Spread and set or use a lodash method.